src/table/components/TableWrapper.jsx

```diff
@@ -63,38 +68,15 @@ export default function TableWrapper(props) {
     });
   };
 
-  useEffect(() => {
-    const memoedWrapper = tableWrapperRef.current;
-    if (!memoedWrapper) return () => {};
-
-    const focusOutCallback = (evt) => handleFocusoutEvent(evt, shouldRefocus, keyboard);
-    memoedWrapper.addEventListener('focusout', focusOutCallback);
-
-    return () => {
-      memoedWrapper.removeEventListener('focusout', focusOutCallback);
-    };
-  }, []);
-
-  useEffect(() => {
-    const memoedContainer = tableContainerRef.current;
-    if (!memoedContainer) return () => {};
-
-    const horizontalScrollCallback = (evt) => handleHorizontalScroll(evt, direction === 'rtl', memoedContainer);
-    memoedContainer.addEventListener('wheel', horizontalScrollCallback);
-
-    return () => {
-      memoedContainer.removeEventListener('wheel', horizontalScrollCallback);
-    };
-  }, [direction]);
+  useFocusListener(tableWrapperRef, shouldRefocus, keyboard);
+  useScrollListener(tableContainerRef, direction);
 
   useEffect(
     () => handleNavigateTop({ tableContainerRef, focusedCellCoord, rootElement }),
-    [tableContainerRef, focusedCellCoord]
+    [tableContainerRef, focusedCellCoord, rootElement]
   );
 
   useDidUpdateEffect(() => {
-    if (!keyboard.enabled) return;
-
     updateFocus({ focusType: keyboard.active ? 'focus' : 'blur', cell: getCellElement(rootElement, focusedCellCoord) });
```

Comment:
Could you add a comment to explain it a bit?

Reply:
when keyboard.enabled is not true, keyboard.active never changes, so this hook will never be triggered. No need for the check

Reply:
no, this hook will never be triggered after renders succeed. It at least runs one time although not necessary.

Reply:
well, that is what `useDidUpdateEffect` should take care of. I just tested and it is broken bc of the useEffect running twice....... Even if keyboard.enabled is true this shouldn't run the first time. And it should be trouble for the other use-case of useDidUpdateEffect. I specifically added the hook since we needed it to not run on first render

Reply:
I type wrong. I mean this hook will be triggered after renders succeed. It at least runs one time although not necessary.

Reply:
aha, right, I recalled. So the useDidUpdateEffect is broken...

Reply:
I could add a counter so it makes sure it skips the first to times it is triggered but it is dirty... I think I read somewhere that it is the react strict mode messing this up, why do we have that? I think you added it? Could we remove it once we convert to typescript?

Reply:
I agree. I think we would get a lot of the same warnings etc from typscript

Reply:
I'll leave this for now since this change doesn't really cause the problem